client/thin-replica-client/src/thin_replica_client.cpp

```diff
@@ -733,13 +736,13 @@ ThinReplicaClient::~ThinReplicaClient() {
     ConcordAssert(subscription_thread_->joinable());
     try {
       subscription_thread_->join();
```

Comment:
`join` will not throw any application-specific exceptions. Instead, it might throw `system_error`.
Meaning, we only need to catch `std::exception` or `system_error` directly.
Therefore, we can remove all the "catch-code" here and for all other `join()` calls.
[https://en.cppreference.com/w/cpp/language/throw](https://en.cppreference.com/w/cpp/language/throw)

Reply:
Thanks for catching, updated.